programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1087,6 +1130,26 @@ static int dns_fallback(const char *name, void *ctx, struct in_addr* addr) {
     return dns_miss_status;
 }
 
+#if _WIN32
+static void interrupt_handler(int sig) {
+    ZITI_LOG(WARN,"Received signal to interrupt");
+    tunnel_interrupted = true;
+    scm_service_stop();
+}
+#endif
+
+static char* normalize_host(char* hostname) {
```

Comment:
this logic is suspect.
1. `len-2` does not point to the last char in string
2. is there a reason to modify passed in string if you're returning an allocated modified copy?
3. this seem to lead to a memory leak

Reply:
I fixed points 1 and 2 in the comment. I have added "model_map_clear(domains, (_free_f) free);" where key is the hostname, so the hostnames will get free'd outside the normalize_host function.